src/runtime/backend.rs

```diff
@@ -0,0 +1,7 @@
+use crate::runtime::module::Module;
+use std::sync::Arc;
+
+pub trait Compiler {
+    /// Compiles a `Module` from WebAssembly binary format
+    fn compile(&self, wasm: &[u8]) -> Result<Arc<Module>, String>;
```

Comment:
Should the Error be an `ErrorKind` here? maybe a `CompilerError`?

Reply:
Yep, I was just throwing strings in the results so I didn't have to think about errors when I first wrote it, but I'd definitely like to have an error type for this.